lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/ProcessCriCallbackHandler.java

```diff
@@ -0,0 +1,344 @@
+import uk.gov.di.ipv.core.library.kmses256signer.KmsEs256Signer;
+import uk.gov.di.ipv.core.library.persistence.item.ClientOAuthSessionItem;
+import uk.gov.di.ipv.core.library.persistence.item.CriOAuthSessionItem;
+import uk.gov.di.ipv.core.library.persistence.item.IpvSessionItem;
+import uk.gov.di.ipv.core.library.service.AuditService;
+import uk.gov.di.ipv.core.library.service.CiMitService;
+import uk.gov.di.ipv.core.library.service.ClientOAuthSessionDetailsService;
+import uk.gov.di.ipv.core.library.service.ConfigService;
+import uk.gov.di.ipv.core.library.service.CriOAuthSessionService;
+import uk.gov.di.ipv.core.library.service.CriResponseService;
+import uk.gov.di.ipv.core.library.service.IpvSessionService;
+import uk.gov.di.ipv.core.library.service.UserIdentityService;
+import uk.gov.di.ipv.core.library.verifiablecredential.exception.VerifiableCredentialResponseException;
+import uk.gov.di.ipv.core.library.verifiablecredential.helpers.VcHelper;
+import uk.gov.di.ipv.core.library.verifiablecredential.service.VerifiableCredentialService;
+import uk.gov.di.ipv.core.library.verifiablecredential.validator.VerifiableCredentialJwtValidator;
+import uk.gov.di.ipv.core.processcricallback.dto.CriCallbackRequest;
+import uk.gov.di.ipv.core.processcricallback.service.CriApiService;
+import uk.gov.di.ipv.core.processcricallback.service.CriCheckingService;
+import uk.gov.di.ipv.core.processcricallback.service.VcStoringService;
+
+import java.text.ParseException;
+import java.util.Map;
+
+import static uk.gov.di.ipv.core.library.journeyuris.JourneyUris.JOURNEY_ERROR_PATH;
+
+public class ProcessCriCallbackHandler
+        implements RequestHandler<CriCallbackRequest, Map<String, Object>> {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final JourneyResponse JOURNEY_ERROR = new JourneyResponse(JOURNEY_ERROR_PATH);
+    private static final String PYI_ATTEMPT_RECOVERY_PAGE_ID = "pyi-attempt-recovery";
+    private static final String PYI_TIMEOUT_RECOVERABLE_PAGE_ID = "pyi-timeout-recoverable";
+    private final ConfigService configService;
+    private final KmsEs256Signer signer;
+    private final CriApiService criApiService;
+    private final VcStoringService vcStoringService;
+    private final CriCheckingService criCheckingService;
+    private final Gpg45ProfileEvaluator gpg45ProfileEvaluator;
+    private final IpvSessionService ipvSessionService;
+    private final AuditService auditService;
+    private final CriOAuthSessionService criOAuthSessionService;
+    private final ClientOAuthSessionDetailsService clientOAuthSessionDetailsService;
+    private final VerifiableCredentialService verifiableCredentialService;
+    private final VerifiableCredentialJwtValidator verifiableCredentialJwtValidator;
+    private final UserIdentityService userIdentityService;
+    private final CiMitService ciMitService;
+    private final CriResponseService criResponseService;
```

Comment:
I think some of these are not used within this class, and are only passed into dependencies - we can probably remove them as instance variables?

E.g. the `criResponseService` is probably only needed in the second constructor